File: android_mobile/src/main/java/com/alexstyl/specialdates/search/BackKeyEditText.java
package com.alexstyl.specialdates.search;

import android.content.Context;
import android.util.AttributeSet;
import android.view.KeyEvent;
import android.widget.EditText;

public class BackKeyEditText extends android.support.v7.widget.AppCompatEditText {
    public BackKeyEditText(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    private OnBackKeyPressedListener listener;

    public void setOnBackKeyPressedListener(OnBackKeyPressedListener listener) {
        this.listener = listener;
    }

    @Override
    public boolean onKeyPreIme(int keyCode, KeyEvent event) {
        if (keyCode == KeyEvent.KEYCODE_BACK && listener.onBackButtonPressed()) {
            return true;
        }
        return super.onKeyPreIme(keyCode, event);
    }
}

File: android_mobile/src/main/java/com/alexstyl/specialdates/search/BackKeyEditText.kt
package com.alexstyl.specialdates.search

import android.content.Context
import android.util.AttributeSet
import android.view.KeyEvent
import android.widget.EditText

class BackKeyEditText(context: Context, attrs: AttributeSet) : android.support.v7.widget.AppCompatEditText(context, attrs) {

    private var listener: OnBackKeyPressedListener? = null

    fun setOnBackKeyPressedListener(listener: OnBackKeyPressedListener) {
        this.listener = listener
    }

    override fun onKeyPreIme(keyCode: Int, event: KeyEvent): Boolean =
            if (keyCode == KeyEvent.KEYCODE_BACK) {
                listener?.onBackButtonPressed() ?: false
            } else super.onKeyPreIme(keyCode, event)
}
